packages/splitter/src/useSplitter.ts

```diff
@@ -0,0 +1,274 @@
+  const [dimension, setDimensionWithCache, dimensionPreCacheRef] = usePrecachedState<number>({
+    defaultValue: defaultValueNow,
+    value: valueNow,
+    onChange
+  });
+
+  useEffect(() => {
+    let direction: 'left' | 'top' = 'left';
+
+    if (orientation === SplitterOrientation.HORIZONTAL) {
+      direction = 'top';
+    }
+    // getBoundingClientRect is in pixel values not unitless
+    startPosRef.current = Number(separatorRef?.current?.getBoundingClientRect?.()[direction]);
+  }, [startPosRef, separatorRef, dimension, orientation]);
+
+  const addDelta = useCallback(
+    (delta: number) => {
+      // this function runs outside the React update loop inside a global mouse event
+      const nextDimension = dimensionPreCacheRef.current! + delta;
+
+      if (nextDimension >= max) {
+        setDimensionWithCache(max);
+      } else if (nextDimension <= min) {
+        setDimensionWithCache(min);
+      } else {
+        setDimensionWithCache(nextDimension);
+      }
+    },
+    [dimensionPreCacheRef, setDimensionWithCache, min, max]
+  );
+
+  const onSplitterMouseMove = useCallback(
+    (event: MouseEvent) => {
+      event.preventDefault();
+      if (orientation === SplitterOrientation.HORIZONTAL) {
+        // event.clientY or X is in pixel values not unitless
+        addDelta(event.clientY - startPosRef.current);
+      } else {
+        // event.clientY or X is in pixel values not unitless
+        addDelta(event.clientX - startPosRef.current);
+      }
+    },
+    [addDelta, orientation]
+  );
+
+  const onMouseMove = composeEventHandlers(props.onMouseMove, onSplitterMouseMove);
+
+  const onSplitterTouchMove = useCallback(
+    (event: TouchEvent) => {
+      const { pageY, pageX } = event.targetTouches[0];
+
+      if (orientation === SplitterOrientation.HORIZONTAL) {
+        // event.clientY or X is in pixel values not unitless
+        addDelta(pageY - startPosRef.current);
+      } else {
+        // event.clientY or X is in pixel values not unitless
+        addDelta(pageX - startPosRef.current);
+      }
+    },
+    [addDelta, orientation]
+  );
+
+  const onTouchMove = composeEventHandlers(props.onTouchMove, onSplitterTouchMove);
+
+  const onMouseUp = composeEventHandlers(props.onMouseUp, (event: MouseEvent) => {
+    event.preventDefault();
+    // must remove global events on transaction finish
+    environment.removeEventListener('mouseup', onMouseUp);
+    environment.removeEventListener('mousemove', onMouseMove);
+  });
+
+  const onTouchEnd = composeEventHandlers(props.onTouchEnd, () => {
+    // must remove global events on transaction finish
+    environment.removeEventListener('touchend', onTouchEnd);
+    environment.removeEventListener('touchmove', onTouchMove);
+  });
+
+  const onMouseDown = composeEventHandlers(props.onMouseDown, (event: React.MouseEvent) => {
+    event.preventDefault();
+    if (type === SplitterType.FIXED) {
```

Comment:
I noticed when clicking on the splitter it jumps really quickly. I think we should provide a fast smooth transition or some sort of snap effect when alternating between the `fixed` sizes.

Reply:
This may be a problem to look into on Component phase. The Containers are minimum viable styling and animation. Good idea though 👍